assets/js/components/surveys/SurveyQuestionSingleSelect.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * SurveyQuestionSingleSelect component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../Button';
+import SurveyHeader from './SurveyHeader';
+import SurveyQuestionSingleSelectChoice from './SurveyQuestionSingleSelectChoice';
+
+// const MAXIMUM_CHARACTER_LIMIT = 100;
+
+/* eslint-disable camelcase */
+
+const SurveyQuestionSingleSelect = ( {
+	question,
+	choices,
+	answerQuestion,
+	dismissSurvey,
+} ) => {
+	const [ value, setValue ] = useState( '' );
+	const [ writeIn, setWriteIn ] = useState( '' );
+
+	const mappedChoices = choices.map( ( choice ) => ( {
+		...choice,
+		answer_ordinal: `${ choice.answer_ordinal }`,
+	} ) );
+
+	const currentSelectedOptionHasWriteIn =
+		!! value &&
+		mappedChoices.filter(
+			( { answer_ordinal, write_in } ) =>
+				answer_ordinal === value && write_in
+		).length > 0;
+
+	const handleSubmit = () => {
+		const answerTextOptionalKey = currentSelectedOptionHasWriteIn
+			? { answer_text: writeIn }
+			: {};
+		answerQuestion( {
+			answer: {
+				answer_ordinal: Number( value ),
+				...answerTextOptionalKey,
+			},
+		} );
+	};
+
+	const isSubmitButtonDisabled =
+		value === '' || ( currentSelectedOptionHasWriteIn && writeIn === '' );
+
+	return (
+		<div className="googlesitekit-single-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+			<div className="googlesitekit-survey__body">
+				{ mappedChoices.map( ( choice, id ) => (
+					<SurveyQuestionSingleSelectChoice
+						key={ id }
+						value={ value }
+						setValue={ setValue }
+						writeIn={ writeIn }
+						setWriteIn={ setWriteIn }
+						choice={ choice }
+					/>
+				) ) }
+			</div>
+			<div className="googlesitekit-survey__footer">
+				<Button
+					onClick={ handleSubmit }
+					disabled={ isSubmitButtonDisabled }
+				>
+					Next
```

Comment:
This should be localised.

Reply:
Same concern as comment here - https://github.com/google/site-kit-wp/pull/3788#discussion_r685917798 - should this be a prop anyway?

Reply:
As mentioned in that PR: let's make it a prop and set it based on the current question ordinal 👍🏻